editors/code/src/inlay_hints.ts

```diff
@@ -63,6 +67,22 @@ const paramHints = {
     }
 };
 
+const chainingHints = {
+    decorationType: vscode.window.createTextEditorDecorationType({
+        after: {
+            color: new vscode.ThemeColor('rust_analyzer.inlayHint'),
+            fontStyle: "normal",
+        }
+    }),
+
+    toDecoration(hint: ra.InlayHint.ChainingHint, conv: lc.Protocol2CodeConverter): vscode.DecorationOptions {
+        return {
+            range: conv.asRange(hint.range),
+            renderOptions: { after: { contentText: ` ${hint.label}` } }
```

Comment:
I did think about this, but it seemed to me that the reason for this colon is for this kind of inlay hint to exactly mimic the type syntax in a `let` statement. In the case of chaining there is no "real" syntax to mimic so I thought the simpler the better. Keen to hear others' opinions though.

Reply:
Type ascription is supported on nightly.

```rust
None: Option<()>;
```

Reply:
I wonder if you can put a real type ascription on nightly between the chains of the method call. I have a strong feeling that it will result in a syntax error, but bruh, the inlay hint must not always resemble a valid syntax.

Reply:
@matklad, @SomeoneToIgnore, @lnicola, any thoughts?

Reply:
I don't think we need a `:` here

Reply:
```rust
error: casts cannot be followed by a method call
 --> src/main.rs:4:13
  |
4 |     let a = vec![]: Vec<u8>.collect::<Vec<u8>>();
  |             ^^^^^^-^^^^^^^^
  |                   |
  |                   help: maybe write a path separator here: `::`
  |
```

Reply:
I like the way the hint is separated from the expression, but I agree that the text we display should sometimes be replaced with something different than `Flatten<Chain<Chain<Once<Map<{unknown}, fn to_string<{unknown}>(&{unknown}) -> String>>, {unknown}>, Once<Option<String>>>>`

Reply:
The ideal hint would be `impl Iterator<Item = String>` when the type is `Flatten<Chain<Chain<Once<Map<{unknown}, fn to_string<{unknown}>(&{unknown}) -> String>>, {unknown}>, Once<Option<String>>>>`.